java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/InputStream.java

```diff
@@ -1620,29 +1611,6 @@ protected String readTypeId(boolean isIndex) {
             }
         }
 
-        protected Class<?> resolveClass(String typeId) {
-            Class<?> cls = null;
-            if (_typeIdCache == null) {
-                _typeIdCache = new HashMap<>(); // Lazy initialization.
```

Comment:
In the existing code, there was a per-InputStream (!) cache for typeId to Class and compactId to Class. I removed them both. The caching, if any, is now performed by the communicator-wide SliceLoader implementations.